### PR TITLE
New Command "BlockedServers"

### PR DESCRIPTION
**Pull request by nupplaphil**
Solves #6245 

```console
console blockedservers - Manage blocked servers
Usage
	bin/console blockedservers [-h|--help|-?] [-v]
	bin/console blockedservers add <server> <reason> [-h|--help|-?] [-v]
	bin/console blockedservers remove <server> [-h|--help|-?] [-v]
```

Examples:

- **`bin/console blockedservers`**
```console
+----------------------------+-----------------+
| Domain                     | Reason          |
+----------------------------+-----------------+
| social.nobodyhasthe.biz    | Illegal content |
| pod.ordoevangelistarum.com | Illegal content |
+----------------------------+-----------------+
```

- **`bin/console blockedservers add test`**
```console
The domain 'test' is now blocked. (Reason: 'blocked')
```

- **`bin/console blockedservers add test "Because it's hard!"`**
```console
The domain 'test' is now blocked. (Reason: 'Because it's hard!')
```

### CODE REVIEW

**Comment by JeroenED** [comment]
1 question: what happens when I block a server that is already blocked with a different reason? Ideally this would just update the reason.

**Comment by nupplaphil** [comment]
> 1 question: what happens when I block a server that is already blocked with a different reason? Ideally this would just update the reason.

Currently, the output would be:
```console
The domain 'test' is already blocked. (Reason: 'blocked')
```

But I think your idea is better :-)

File: src/Core/Console.php [hunk 1]
```diff
@@ -14,6 +14,7 @@ class Console extends \Asika\SimpleConsole\Console
 	protected $customHelpOptions = ['h', 'help', '?'];
 
 	protected $subConsoles = [
+		'blockedservers'         => __NAMESPACE__ . '\Console\BlockedServers',
```
Comment (MrPetovan) [inline src/Core/Console.php]: The other block command is called `globalcommunityblock`, maybe we should name this `serverblock`?

File: src/Core/Console/ServerBlock.php [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+			}
+		}
+
+		if (!$found) {
+			$this->out(sprintf("The domain '%s' is not blocked.", $domain));
+			return 1;
+		}
+
+		if ($config->set('system', 'blocklist', $newBlockList)) {
+			$this->out(sprintf("The domain '%s' is not more blocked", $domain));
```
Comment (JeroenED) [inline src/Core/Console/ServerBlock.php]: Shouldn't this be "The domain '%s' is not blocked anymore"? @MrPetovan

Comment (MrPetovan) [reply src/Core/Console/ServerBlock.php]: Ow, yeah. This is correct but awkward.

Comment (nupplaphil) [reply src/Core/Console/ServerBlock.php]: yoda: "English is my best language not.  Herh herh herh."

**Comment by hoergen** [comment]
Is the http:// or https:// automatically stripped? Because I ran into this problem when using it on the GUI. I think we should cover this.

**Pull request merged**